--- icu/source/test/intltest/intltest.cpp	2013-01-11 01:22:00.000000000 +0100
+++ icu/source/test/intltest/intltest.cpp	2018-06-15 09:17:32.714205868 +0200
@@ -1425,7 +1425,11 @@
     if(ctest_xml_fini())
       return 1;
 
-    return major.getErrors();
+    if(!warnOnMissingData) {
+      return major.getErrors();
+    } else {
+      return 0;
+    }
 }
 
 const char* IntlTest::loadTestData(UErrorCode& err){
